How to Build a WordPress Theme in WordPress Playground

karthick-murugan opened the pull request:
Fixes https://github.com/WordPress/wordpress-playground/issues/1257

This PR adds a new documentation page that explains how to build a WordPress theme using [WordPress Playground](https://playground.wordpress.net/). The guide is aimed at theme developers and contributors looking for a fast, browser-based workflow to build, test, and share themes without needing a local setup.

✅ What’s Included

- How to create a new Playground site (locally or in the browser)
- How to preinstall development tools like Create Block Theme, Gutenberg, and Theme Check
- Detailed steps to:
  - 🚀 Install a starter theme (via GitHub or ZIP)
  - 📥 Import demo content
  - 🔧 Build and customize using block tools
  - 📤 Export themes (ZIP & GitHub)
  - 🔁 Re-import/export for review and iteration
  - 🌐 Load themes from GitHub pull requests

bgrgicak commented on packages/docs/site/docs/main/how-to-create-a-theme-using-playground.md:
```diff
@@ -0,0 +1,164 @@
+
+# How to Build a WordPress Theme in WordPress Playground
+
+WordPress Playground is a powerful tool that lets you build, preview, and export WordPress themes—all in your browser or locally via `wp-now`.
```
Let's use Playground CLI in examples because that's the tool we are focused on.

bgrgicak reviewed (commented):
How does this docs page fit in with the [existing theme development section](https://wordpress.github.io/wordpress-playground/guides/for-theme-developers/#theme-development)?

karthick-murugan commented:
> How does this docs page fit in with the [existing theme development section](https://wordpress.github.io/wordpress-playground/guides/for-theme-developers/#theme-development)?

@bgrgicak May be we can enhance the theme development section with our new page content? More insights will be helpful

bgrgicak commented:
This could become a _Block theme development using the Playground Website_ section on the existing page.

Regarding the content you created, you were precise in explaining how to configure Playground, which is nice and makes it clear for the user. With Playground, we can take it a step further and replace these setup instructions with a Playground link that contains a Blueprint, so users can have a Playground ready to go in one click. 🙂

karthick-murugan commented:
> This could become a _Block theme development using the Playground Website_ section on the existing page.

> Regarding the content you created, you were precise in explaining how to configure Playground, which is nice and makes it clear for the user. With Playground, we can take it a step further and replace these setup instructions with a Playground link that contains a Blueprint, so users can have a Playground ready to go in one click. 🙂

Out of the 8 steps which I have added, what are all the points that can be removed and replaced?

Also adding this json link ( https://playground.wordpress.net/?blueprint=https%3A%2F%2Fgithub.com%2Fwordpress%2Fblueprints%2Fblob%2Ftrunk%2Fblueprints%2Fcreate-block-theme%2Fblueprint.json ) to playground does not activate the "create block theme" in playground. What mistake i am doing here?

cc @bph

bph commented:
>Also adding this json link ( https://playground.wordpress.net/?blueprint=https%3A%2F%2Fgithub.com%2Fwordpress%2Fblueprints%2Fblob%2Ftrunk%2Fblueprints%2Fcreate-block-theme%2Fblueprint.json ) to playground does not activate the "create block theme" in playground. What mistake i am doing here?

@karthick-murugan two things: 

- The query parameter is `blueprint-url`
- you need to use `raw.gihubusercontent.com` to reference a blueprint.json file on GitHub (see #2148)

`https://playground.wordpress.net/?blueprint-url=https://raw.githubusercontent.com/WordPress/blueprints/refs/heads/trunk/blueprints/create-block-theme/blueprint.json`

[Will work. Try it](https://playground.wordpress.net/?blueprint-url=https://raw.githubusercontent.com/WordPress/blueprints/refs/heads/trunk/blueprints/create-block-theme/blueprint.json)

adamziel commented:
@bph do you think this one is good to go?

bph commented:
@adamziel @karthick-murugan  I'll review again tomorrow first thing.

bph commented on packages/docs/site/docs/main/how-to-create-a-theme-using-playground.md:
```diff
@@ -0,0 +1,160 @@
+3. Choose WordPress and PHP versions if needed
+4. Click **"Launch"**
+
+### Locally with `wp-now`
```
I would remove the wp-now reference and instructions, as it's going to be replaced by Playground CLI. The commands not correct anyway. 

It would be `npx @wp-now/wp-now start`

bph commented on packages/docs/site/docs/main/how-to-create-a-theme-using-playground.md:
```diff
@@ -0,0 +1,160 @@
+-   [Gutenberg](https://wordpress.org/plugins/gutenberg/)
+-   [Theme Check](https://wordpress.org/plugins/theme-check/)
+
+Directly use playground links to load plugins: https://playground.wordpress.net/#%7B%22plugins%22:%5B%22hello-dolly%22,%22gutenberg%22%5D,%20%22login%22:%20true,%20%22landingPage%22:%20%22/wp-admin/plugins.php%22%20%7D
```
This Playground link only installs the hello-dolly plugin again and the Gutenberg plugin but not the Create-Block-Theme plugin.

bph commented on packages/docs/site/docs/main/how-to-create-a-theme-using-playground.md:
```diff
@@ -0,0 +1,160 @@
+2. Click **Choose File** and select your `.zip` theme file
+3. Click **Install Now**
+4. Once installed, click **Activate**
+
```
These built up could be solved much faster with a blueprint and that would have the advantage that it can be called repeatedly with only one click and a step.

adamziel commented:
This PR seems stale so I'll close it. Feel free to reopen at any point but also no pressure – it can stay closed and that's fine.
